ui/lib/sync/addon/routes/secrets/destinations/destination/secrets.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
+
+import type StoreService from 'vault/services/store';
+import SyncDestinationModel from 'vault/vault/models/sync/destination';
+
+interface SyncDestinationSecretsRouteParams {
+  page: string;
+}
+
+export default class SyncDestinationSecretsRoute extends Route {
+  @service declare readonly store: StoreService;
+
+  model(params: SyncDestinationSecretsRouteParams) {
+    const destination = this.modelFor('secrets.destinations.destination') as SyncDestinationModel;
+    return hash({
+      destination,
+      associations: this.store.lazyPaginatedQuery('sync/association', {
+        responsePath: 'data.keys',
+        page: Number(params.page) || 1,
```

Comment:
What are your thoughts on creating a controller file and putting the default query params in there? I did some [reading](https://guides.emberjs.com/release/routing/query-params/) on this and saw that by defining a default value for the query params in the controller we don't need to parse the `params.page` value using Number or JSON.parse since the controller will handle that. I've also noticed from the [ember docs](https://guides.emberjs.com/release/routing/query-params/) query params and default values are typically set on the controller. I'm not sure what the exact pattern is since creating a controller file just for default params seems like it may not be necessary.

Reply:
🤔 I think the issue with defining default params on the controller is then we can't remove them. So they don't clear when we expect them to. This was an issue on KV and it's why I removed them from the controller and only relied on the default page to be returned by lazyPaginatedQuery. If I'm incorrect, or there's another flow, I'd be curious how to implement.

Reply:
Actually @kiannaquach I was wrong 😞. You are correct about the value being cast to whatever the datatype is in the default value! I tested it quick to verify 🎉 .

Reply:
@Monkeychip to your point [query params are sticky by default](https://guides.emberjs.com/release/routing/query-params/#toc_sticky-query-param-values)